### PR TITLE
Dataflow: Add default taint sanitizer guard

### PR DESCRIPTION
**Pull request by owen-mc**
This is used in codeql-go.

### CODE REVIEW

**Review by hvitved — DISMISSED** [review]
LGTM

File: java/ql/lib/semmle/code/java/dataflow/internal/TaintTrackingUtil.qll [hunk 1]
```diff
@@ -113,6 +113,13 @@ private module Cached {
     node.getEnclosingCallable().getDeclaringType() instanceof NonSecurityTestClass or
     node.asExpr() instanceof ValidatedVariableAccess
   }
+
+  /**
+   * Holds if `guard` should be a sanitizer guard in all global taint flow configurations
+   * but not in local taint.
+   */
+  cached
```
Comment (aschackmull) [inline java/ql/lib/semmle/code/java/dataflow/internal/TaintTrackingUtil.qll:121]: Caching this seems overkill.

Comment (owen-mc) [reply java/ql/lib/semmle/code/java/dataflow/internal/TaintTrackingUtil.qll]: Oops, I think I just copied what was going on for defaultTaintSanitizer and didn't engage my brain enough.

**Review by aschackmull — APPROVED** [review]
LGTM

**Pull request merged**